### DISCUSSION TITLE
Migrating to Csla 6 - DI Scope management

### DISCUSSION THREAD

**Discussion by ajohnstone-ks** [MarimerLLC/csla#3599] [Questions]
In Csla 5, we implemented `IInterceptDataPortal` so that we can create an `IServiceScope` that lives for the lifetime of a "root" level data portal call.  To do that it was setting `ApplicationContext.ServiceProviderScope`

That doesn't work anymore, because `ApplicationContext` is no longer static, and no longer has that property.  

Where do I need to plug into now to handle the scope creation/disposal?

**Comment by rockfordlhotka** [comment]
You should no longer need to manage the DI scope yourself. Because CSLA 6+ fully incorporates DI, there is a DI scope created/disposed for every server-side data portal operation.

It _might_ be that this doesn't fully work until CSLA 7 - we did fix some bugs around this and I don't remember specifically when.

(as an aside, I'd personally just skip CSLA 6 and go for 7 - there are few breaking changes, and many improvements!)

**Reply by ajohnstone-ks** [reply]
Ok, that is good to know.  I still have about 4000 compiler errors to fix before I can start testing anything though. 😆

**Reply by Chicagoan2016** [reply]
Only 4,000!

**Reply by ajohnstone-ks** [reply]
@Chicagoan2016 That's just in the BO assembly only; i'm not sure what else is broken outside of that :)
